src/recorder.ts

```diff
@@ -61,20 +61,59 @@ export async function isSubmitButton(
   return isSubmitButtonResponse.result.value;
 }
 
+type AXNode = protocol.Protocol.Accessibility.AXNode;
+type CDPSession = puppeteer.CDPSession;
+
+// We check that a selector uniquely selects an element by querying the
+// selector and checking that all found elements are in the subtree of the
+// target.
+async function checkUnique(
+  client: CDPSession,
+  ignored: AXNode[],
+  name?: string,
+  role?: string
+) {
+  const { root } = await client.send('DOM.getDocument', { depth: 0 });
+  const checkName = await client.send('Accessibility.queryAXTree', {
+    backendNodeId: root.backendNodeId,
+    accessibleName: name,
+    role: role,
+  });
+  const checkNameMinusTargetTree = checkName.nodes.filter(
+    (axNode) =>
+      !ignored.find(
+        (otherAxNode) =>
+          axNode.backendDOMNodeId === otherAxNode.backendDOMNodeId
+      )
+  );
```

Comment:
Instead of repeatedly calling `Array#find()` (`O(n)`) potentially `n` times (for each `filter` callback for `n` nodes), we could do a single pass up-front to create a `Set` of `backendDOMNodeId`s. In the `filter` callback we could then simply check if the set contains the ID, which is `O(1)`. WDYT?

Then again, in the common case we expect only a few nodes to be returned, so maybe it doesn't matter too much. WDYT?

Reply:
Yep the list of nodes should be fairly small, but let's go with the set of `backendDOMNodeId`s. I'll update.